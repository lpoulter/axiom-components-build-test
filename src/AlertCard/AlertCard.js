import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import AlertIcon from '../AlertIcon/AlertIcon';
import Card from '../Card/Card';
import CardContent from '../Card/CardContent';
import Icon from '../Icon/Icon';
import Link from '../Typography/Link';
import './AlertCard.css';

export default class AlertCard extends Component {
  static propTypes = {
    /** Content displayed next to the AlertIcon */
    children: PropTypes.node.isRequired,
    /** An optional callback that when given adds a removable cross */
    onRemoveClick: PropTypes.func,
    /** Shade of the background color */
    shade: PropTypes.oneOf(['shade-1', 'shade-2', 'shade-3', 'shade-4']),
    /** Increases/decreases the size of the card */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    /** Type of AlertBar that affects the coloring and icon */
    type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  };

  static defaultProps = {
    size: 'medium',
    type: 'info',
  };

  render() {
    const { children, size, type, onRemoveClick, ...rest } = this.props;
    const classes = classnames('ax-alert-card', `ax-alert-card--${size}`);

    return (
      <Card { ...rest } size={ size }>
        <CardContent size={ size }>
          <div className={ classes }>
            <div className="ax-alert-card__icon">
              <AlertIcon type={ type } />
            </div>

            <div className="ax-alert-card__content">
              { children }
            </div>

            { onRemoveClick && (
              <div className="ax-alert-card__remove">
                <Link onClick={ onRemoveClick } style="subtle">
                  <Icon name="cross" />
                </Link>
              </div>
            ) }
          </div>
        </CardContent>
      </Card>
    );
  }
}
