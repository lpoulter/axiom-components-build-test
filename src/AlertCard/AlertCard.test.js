import React from 'react';
import renderer from 'react-test-renderer';
import AlertCard from './AlertCard';

const getComponent = (props = {}) =>
  renderer.create(
    <AlertCard { ...props }>Lorem</AlertCard>
  );

describe('AlertCard', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with onRemoveClick', () => {
    const component = getComponent({ onRemoveClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with shade', () => {
    ['shade-1', 'shade-2', 'shade-3', 'shade-4'].forEach((shade) => {
      it(shade, () => {
        const component = getComponent({ shade });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with size', () => {
    ['small', 'medium', 'large'].forEach((size) => {
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with type', () => {
    ['success', 'warning', 'error', 'info'].forEach((type) => {
      it(type, () => {
        const component = getComponent({ type });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

});
