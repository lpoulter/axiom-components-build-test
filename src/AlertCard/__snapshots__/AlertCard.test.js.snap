// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertCard renders with defaultProps 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with onRemoveClick 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
      <div
        className="ax-alert-card__remove"
      >
        <a
          className="ax-link ax-link--style-subtle"
          onClick={[Function]}
        >
          <svg
            className="ax-icon ax-icon--cross"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with shade shade-1 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with shade shade-2 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-2 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with shade shade-3 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-3 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with shade shade-4 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-4 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with size large 1`] = `
<div
  className="ax-card ax-card--size-large ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-large"
  >
    <div
      className="ax-alert-card ax-alert-card--large"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with size medium 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with size small 1`] = `
<div
  className="ax-card ax-card--size-small ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
  >
    <div
      className="ax-alert-card ax-alert-card--small"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with type error 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--error"
        >
          <svg
            className="ax-icon ax-icon--warning"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with type info 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
        >
          <svg
            className="ax-icon ax-icon--information"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with type success 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--success"
        >
          <svg
            className="ax-icon ax-icon--tick"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertCard renders with type warning 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-card__content ax-card__content--separator-solid ax-card__content--size-medium"
  >
    <div
      className="ax-alert-card ax-alert-card--medium"
    >
      <div
        className="ax-alert-card__icon"
      >
        <div
          className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--warning"
        >
          <svg
            className="ax-icon ax-icon--warning"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            style={
              Object {
                "height": "1rem",
                "width": "1rem",
              }
            }
            viewBox="0 0 16 16"
          />
        </div>
      </div>
      <div
        className="ax-alert-card__content"
      >
        Lorem
      </div>
    </div>
  </div>
</div>
`;
